src/lib/store.js

```diff
@@ -407,72 +271,120 @@ export default class Store {
 	};
 
 	selectPurpose = (purposeId, isSelected) => {
-		const {selectedPurposeIds} = this.vendorConsentData;
+		const {purposeConsents} = this.tcModel;
 		if (isSelected) {
-			selectedPurposeIds.add(purposeId);
-		}
-		else {
-			selectedPurposeIds.delete(purposeId);
+			purposeConsents.set(purposeId);
+		} else {
+			purposeConsents.unset(purposeId);
 		}
 		this.storeUpdate();
 	};
 
 	selectAllPurposes = (isSelected) => {
-		const {purposes = []} = this.vendorList || {};
-		const operation = isSelected ? 'add' : 'delete';
-		purposes.forEach(({id}) => this.vendorConsentData.selectedPurposeIds[operation](id));
+		const operation = isSelected ? 'setAllPurposeConsents' : 'unsetAllPurposeConsents';
+		this.tcModel[operation]();
 		this.storeUpdate();
 	};
 
-	selectStandardPurpose = (purposeId, isSelected) => {
-		const {selectedStandardPurposeIds} = this.publisherConsentData;
+	selectPurposeLegitimateInterests = (purposeId, isSelected) => {
+		const {purposeLegitimateInterests} = this.tcModel;
 		if (isSelected) {
-			selectedStandardPurposeIds.add(purposeId);
-		}
-		else {
-			selectedStandardPurposeIds.delete(purposeId);
+			purposeLegitimateInterests.set(purposeId);
+		} else {
+			purposeLegitimateInterests.unset(purposeId);
 		}
 		this.storeUpdate();
 	};
 
-	selectAllStandardPurposes = (isSelected) => {
-		const {purposes = []} = this.vendorList || {};
-		const operation = isSelected ? 'add' : 'delete';
-		purposes.forEach(({id}) => this.publisherConsentData.selectedStandardPurposeIds[operation](id));
+	selectAllPurposesLegitimateInterests = (isSelected) => {
+		const operation = isSelected ? 'setAllPurposeLegitimateInterests' : 'unsetAllPurposeLegitimateInterests';
+		this.tcModel[operation]();
 		this.storeUpdate();
 	};
 
-	selectCustomPurpose = (purposeId, isSelected) => {
-		const {selectedCustomPurposeIds} = this.publisherConsentData;
+	selectSpecialFeature = (specialFeatureId, isSelected) => {
```

Comment:
For consistency reason probably it would be better to change name of this function
```suggestion
	selectSpecialFeatureOptins = (specialFeatureId, isSelected) => {
```
Or alternatively please consider changing name of `selectAllSpecialFeatureOptins` function to `selectAllSpecialFeatures`

Reply:
Renamed to: selectSpecialFeatureOptins